### PR TITLE
BUG/REF: unstack with EA dtypes

### PR DESCRIPTION
**Pull request by jbrockmendel**
1) `ser.unstack(args)` should behave like `ser.to_frame().unstack(args).droplevel(level=0, axis=1)`.  This fails on master, is fixed by this PR.
2) `reshape._unstack_extension_series` currently re-implements logic that is in `ExtensionBlock._unstack`.  This de-duplicates by dispatching.
3) We currently transpose in cases where we shouldn't.  This implements `DataFrame._can_fast_transpose` to avoid that.
4) `test_astype_object_series` has what looks like a typo that is making us test not-the-intended-thing; @TomAugspurger can you confirm im reading this right?
5) The dtype comparisons in `test_astype_object_frame` currently raise, are commented out for exposition, should be fixed in a separate PR.

### CODE REVIEW

File: pandas/tests/extension/base/casting.py [hunk 1]
```diff
@@ -10,10 +10,22 @@ class BaseCastingTests(BaseExtensionTests):
     """Casting to and from ExtensionDtypes"""
 
     def test_astype_object_series(self, all_data):
-        ser = pd.Series({"A": all_data})
+        ser = pd.Series(all_data, name="A")
```
Comment (TomAugspurger) [inline pandas/tests/extension/base/casting.py:13]: Yep, that was a bug.

**Review by TomAugspurger — COMMENTED** [review]
Thanks. How do the unstack asvs look here?

**Comment by jbrockmendel** [comment]
> How do the unstack asvs look here?

with "-b reshape" no change

**Comment by jreback** [comment]
thanks nice de-duplicating

**Pull request merged**